torchtitan/experiments/flux/dataset/flux_dataset.py

```diff
@@ -192,52 +192,60 @@ def _get_data_iter(self):
         return it
 
     def __iter__(self):
+        dataset_iterator = self._get_data_iter()
         while True:
-            for sample in self._get_data_iter():
-                # Use the dataset-specific preprocessor
-                sample_dict = self._data_processor(
-                    sample,
-                    self._t5_tokenizer,
-                    self._clip_tokenizer,
-                    output_size=self.job_config.training.img_size,
-                )
-
-                # skip low quality image or image with color channel = 1
-                if sample_dict["image"] is None:
-                    logger.warning(
-                        f"Low quality image {sample['__key__']} is skipped in Flux Dataloader"
-                    )
+            try:
+                sample = next(dataset_iterator)
+            except StopIteration:
+                if not self.infinite:
+                    logger.warning(f"Dataset {self.dataset_name} has run out of data")
+                    break
+                else:
+                    # Reset offset for the next iteration if infinite
+                    self._sample_idx = 0
+                    logger.warning(f"Dataset {self.dataset_name} is being re-looped")
```

Comment:
Since users specify `infinite`, this is expected. So using `warning` is too strong, maybe `info()`?